includes/admin/load.php

```diff
@@ -77,7 +77,16 @@ function perflab_render_settings_page() {
  * @param string $hook_suffix The current admin page.
  */
 function perflab_admin_pointer( $hook_suffix ) {
+	$current_user = get_current_user_id();
+
 	if ( ! in_array( $hook_suffix, array( 'index.php', 'plugins.php' ), true ) ) {
+
+		// Do not show on the settings page and dismiss the pointer.
+		if ( ! empty( $_GET['page'] ) && PERFLAB_SCREEN === $_GET['page'] ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			$dismissed[] = 'perflab-admin-pointer';
+			update_user_meta( $current_user, 'dismissed_wp_pointers', implode( ',', $dismissed ) );
```

Comment:
wouldn't this overwrite existing dismissed pointers? This probably needs to get the existing list first and append this pointer if it doesn't exist. The core function `wp_ajax_dismiss_wp_pointer` has a good boilerplate for doing this.

Reply:
I think this would be the case if the `$dismissed` assignment line from below is also moved up after setting `$current_user`.

Reply:
thanks @adamsilverstein, @mukeshpanchal27 e @westonruter, made a new commit